### PR TITLE
search: expand and/or query visitor

### PR DESCRIPTION
**Pull request by rvantonder**
I might expand/refactor this more later. Right now it's a means to an end for and/or eval.

Tests: `containsPattern` depends on this being correct, so I'm punting on additional tests for now.

### CODE REVIEW

File: internal/search/query/parser.go [hunk 1]
```diff
@@ -207,31 +207,53 @@ func (p *parser) ParseParameter() Parameter {
...
 	case Operator:
 		f(v)
-		for _, n := range v.Operands {
-			visit(n, f)
+		Visit(v.Operands, f)
+	}
+}
+
+// Visit calls f on all nodes rooted at nodes.
+func Visit(nodes []Node, f func(node Node)) {
```
Comment (unknwon) [inline internal/search/query/parser.go]: I couldn't differ this with `VisitNode` based on the docstring, so I think one of them has inappropriate docstring?

Comment (keegancsmith) [reply internal/search/query/parser.go]: I agree, this also doesn't seem like a very useful function to export. I would suggest `VisitNode` be called `Visit` and this function not be exported (and be called `visitNodes`)

Actually, do you expect the main use of the clients of this package to normally have a `[]Node` and then they use these helper functions? Should we maybe not represent `[]Node` as a type of `Node` like `type And []Node`

Comment (rvantonder) [reply internal/search/query/parser.go]: > I couldn't differ this with VisitNode based on the docstring, so I think one of them has inappropriate docstring?

The one ends with `node.` the other `nodes.`. The difference is one operates on a list of nodes, the other on a single node.

> Actually, do you expect the main use of the clients of this package to normally have a []Node and then they use these helper functions?

I'm still going back and forth on this and haven't decided. Basically, whether we define, e.g.,

`type tree []Node` or `type tree Node`

There are pros and cons to both (not going to go into that right now). I feel like I don't need to make that decision immediately :-) which is why I prefaced this PR with:

> I might expand/refactor this more later. Right now it's a means to an end for and/or eval.

Basically, I want time to think about a good interface here and I'm learning things as I go along. I will come back to it when I have time to make it a focus.

Comment (rvantonder) [reply internal/search/query/parser.go]: Also, not that I don't appreciate feedback--it'd be great to get into this more once I put up some dedicated interfaces for visitors/mappers.

Comment (keegancsmith) [reply internal/search/query/parser.go]: > > I couldn't differ this with VisitNode based on the docstring, so I think one of them has inappropriate docstring?

> The one ends with node. the other nodes.. The difference is one operates on a list of nodes, the other on a single node.

I think you may have missed this. This other function is called `Visit`, not `VisitNode`. Hence the confusion/inconsistency.
